### PR TITLE
Array sums

### PR DESCRIPTION
**Pull request by Sreedevi-27**

### CODE REVIEW

File: src/main/java/algorithms/leetcode/easy/SquareRoot.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Constraints:
+0 <= x <= 231 - 1
+ */
+
+public class SquareRoot {
+    public static int mySqrt(int x) {
+        long sqrt = 0;
+        if(x == 0 || x == 1) return x;
+        for(long i = 2;i<=x;i++){
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/easy/SquareRoot.java:27]: can be refactored

File: src/main/java/algorithms/leetcode/medium/SetMatrixZero.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                if(row[i] || column[j])
+                    matrix[i][j] = 0;
+            }
+        }
+        return matrix;
+    }
+}
+
+/*
+Time complexity : O(r*c) + O(r*c) => O(r*c ^ 2)
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/SetMatrixZero.java:80]: how is this squared? O(r*c) + O(r*c) => O(2r*c) right?

Comment (sharathr25) [reply src/main/java/algorithms/leetcode/medium/SetMatrixZero.java:80]: approach 1 takes less time and space

File: src/main/java/algorithms/leetcode/medium/SetMatrixZero.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    /*
+    APPROACH 1 :
+       int r = matrix.length;
+       int c = matrix[0].length;
+       Set<Integer> row = new HashSet<>();
+       Set<Integer> column = new HashSet<>();
+       for(int i=0; i<r; i++){
+           for(int j=0; j<c; j++){
+               int element = matrix[i][j];
+               if(element==0){
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/SetMatrixZero.java:32]: `element` is getting used only here, so probably replace it with `matrix[i][j]`

File: src/main/java/algorithms/leetcode/easy/SquareRoot.java (hunk 1 above)
Comment (Sreedevi-27) [reply src/main/java/algorithms/leetcode/easy/SquareRoot.java:27]: Yes... If I change it to x/2, it is working fine. But in leetcode, the time taken is more
1. For approach i<=x, it is showing as 23ms  - https://leetcode.com/problems/sqrtx/submissions/1075172597/
2. For approach i<=x/2, it is showing as 57ms - https://leetcode.com/problems/sqrtx/submissions/1077818341/

**Review by sharathr2510 — APPROVED** [review]

**Pull request merged**